### PR TITLE
Details api

### PR DESCRIPTION
**Pull request by hedita**

### CODE REVIEW

File: assets/js/details.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+  const params = new URLSearchParams(window.location.search);
+  const movieId = params.get("id");
+  const apiKey = "9f120f0e95e5cedf05a78abda9ef8477";
+  
+  async function getMovieDetails () {
+   const result = await fetch(`https://api.themoviedb.org/3/movie/${movieId}?api_key=${apiKey}`);
+   const resultJson = await result.json();
+   const {results: movie} = resultJson;
```
Comment (blacksrc) [inline assets/js/details.js]: you can skip defining `resultJson`.

File: assets/js/details.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+   console.log(resultJson);
```
Comment (blacksrc) [inline assets/js/details.js]: Please remove this.

File: assets/js/details.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+   listHtml = "";
+
+   listHtml += `<img id="movie-img" class="img" src="https://image.tmdb.org/t/p/w342/${resultJson.backdrop_path}" />
+   <figcaption class="details">
+     <h2 id="movie-name" class="movie-name"${resultJson.title}></h2>
+     <p id="rate" class="rate">${resultJson.vote_average}</p>
+     <p id="over-view" class="summary">${resultJson.overview}</p>
+   </figcaption>` 
+
+   detailsContainer.innerHTML = listHtml;
```
Comment (blacksrc) [inline assets/js/details.js]: `listHtml` is redundant.Think about how you can just remove the variable

File: assets/js/details.js (hunk 1 above)
Comment (blacksrc) [reply assets/js/details.js]: Also destructure whatever is necessary here annd use it there (`listHtml`).

File: assets/js/main.js [hunk 4]
```diff
@@ -1,15 +1,14 @@
 async function getMovies() {
-  const result = await fetch("http://localhost:8080/");
+  const result = await fetch("https://api.themoviedb.org/3/movie/top_rated?api_key=9f120f0e95e5cedf05a78abda9ef8477&language=en-US&page=1");
```
Comment (blacksrc) [inline assets/js/main.js]: you already have `apiKey` in a variable.

File: assets/js/main.js [hunk 5]
```diff
@@ -1,15 +1,14 @@
...
   const {results: movies} = resultJson;
   const movieList = document.getElementById("movie-list");
-  const imgSrc = "./assets/test.jpg";
   let listHtml = "";
 
   movies.forEach(movie => {
-    listHtml += `<li class="movie-item">
-    <a href="./movie.html">
-      <img id="img" class="img" src=${imgSrc}>
+    listHtml += `<li id="movie-item" class="movie-item">
```
Comment (blacksrc) [inline assets/js/main.js]: Remove all `id`s that are defined in the loop. The reason is that then you will have multiple identical ids in the DOM. Which is not valid.

File: assets/js/main.js [hunk 6]
```diff
@@ -1,15 +1,14 @@
 async function getMovies() {
-  const result = await fetch("http://localhost:8080/");
+  const result = await fetch("https://api.themoviedb.org/3/movie/top_rated?api_key=9f120f0e95e5cedf05a78abda9ef8477&language=en-US&page=1");
   const resultJson = await result.json();
   const {results: movies} = resultJson;
   const movieList = document.getElementById("movie-list");
-  const imgSrc = "./assets/test.jpg";
   let listHtml = "";
 
   movies.forEach(movie => {
```
Comment (blacksrc) [inline assets/js/main.js]: Can you destructrue and re-name (at the same time) properties of the movie object here? 👀 
```
  movies.forEach(movie => {
```

File: assets/js/main.js (hunk 5 above)
Comment (hedita) [reply assets/js/main.js]: Done.

File: assets/js/details.js (hunk 3 above)
Comment (hedita) [reply assets/js/details.js]: Done.

File: assets/js/details.js (hunk 2 above)
Comment (hedita) [reply assets/js/details.js]: Done.

File: assets/js/details.js (hunk 1 above)
Comment (hedita) [reply assets/js/details.js]: Done.

File: assets/js/details.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+   const movieName = document.getElementById("movie-name");
+   const movieImg = document.getElementById("movie-img");
+   const movieRate = document.getElementById("rate");
+   const movieOverView = document.getElementById("over-view");
+   const detailsContainer = document.getElementById("details-container");
+
+   document.body.style = ` background-image: linear-gradient(
+      var(--primary-gradient-color),
+      var(--secondery-gradient-color)),
+      url("https://image.tmdb.org/t/p/w342/${resultJson.poster_path}");`
```
Comment (blacksrc) [inline assets/js/details.js]: I'll put a voice for this.

File: assets/js/main.js (hunk 4 above)
Comment (hedita) [reply assets/js/main.js]: Done.

File: assets/js/details.js (hunk 7 above)
Comment (hedita) [reply assets/js/details.js]: Ok.

File: assets/js/details.js (hunk 2 above)
Comment (blacksrc) [reply assets/js/details.js]: Never mind 😅

File: assets/js/details.js (hunk 7 above)
Comment (blacksrc) [reply assets/js/details.js]: Never mind 😅

File: assets/js/details.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
+  const params = new URLSearchParams(window.location.search);
+  const movieId = params.get("id");
+  const apiKey = "9f120f0e95e5cedf05a78abda9ef8477";
+  
+  async function getMovieDetails () {
+   const result = await fetch(`https://api.themoviedb.org/3/movie/${movieId}?api_key=${apiKey}`);
+   const resultJson = await result.json();
```
Comment (blacksrc) [inline assets/js/details.js]: destructure and rename please.

File: assets/js/details.js [hunk 9]
```diff
@@ -4,7 +4,7 @@
   
   async function getMovieDetails () {
    const result = await fetch(`https://api.themoviedb.org/3/movie/${movieId}?api_key=${apiKey}`);
-   const resultJson = await result.json();
+   const movie = await result.json();
```
Comment (blacksrc) [inline assets/js/details.js]: It's better now. But, You can destructre the properties from movie directly here.

File: assets/js/details.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  const apiKey = "9f120f0e95e5cedf05a78abda9ef8477";
+  
+  async function getMovieDetails () {
+   const result = await fetch(`https://api.themoviedb.org/3/movie/${movieId}?api_key=${apiKey}`);
+   const movie = await result.json();
+
+   const movieName = document.getElementById("movie-name");
+   const movieImg = document.getElementById("movie-img");
+   const movieRate = document.getElementById("rate");
+   const movieOverView = document.getElementById("over-view");
```
Comment (takrami) [inline assets/js/details.js]: where did you use these variables?

File: assets/js/details.js (hunk 9 above)
Comment (hedita) [reply assets/js/details.js]: Done.

File: assets/js/details.js (hunk 10 above)
Comment (hedita) [reply assets/js/details.js]: Oh! You are right.
Removed.

File: assets/js/details.js (hunk 8 above)
Comment (hedita) [reply assets/js/details.js]: Done.

File: assets/js/main.js (hunk 6 above)
Comment (hedita) [reply assets/js/main.js]: Done.

File: assets/js/main.js [hunk 11]
```diff
@@ -1,23 +1,24 @@
...
-  const result = await fetch("http://localhost:8080/");
+  const result = await fetch(`https://api.themoviedb.org/3/movie/top_rated?api_key=${apiKey}&language=en-US&page=1`);
   const resultJson = await result.json();
   const {results: movies} = resultJson;
   const movieList = document.getElementById("movie-list");
-  const imgSrc = "./assets/test.jpg";
   let listHtml = "";
 
-  movies.forEach(movie => {
+  movies.forEach(({id,poster_path: img,title,release_date}) => {
```
Comment (blacksrc) [inline assets/js/main.js]: Rename the variables in a camel case format while destructuring.

File: assets/js/details.js [hunk 12]
```diff
@@ -0,0 +1,23 @@
+  const params = new URLSearchParams(window.location.search);
+  const movieId = params.get("id");
+  const apiKey = "9f120f0e95e5cedf05a78abda9ef8477";
+  const detailsContainer = document.getElementById("details-container");
+  
+  async function getMovieDetails () {
+   const result = await fetch(`https://api.themoviedb.org/3/movie/${movieId}?api_key=${apiKey}`);
+   const {poster_path: img, title, vote_average, overview} = await result.json();
```
Comment (blacksrc) [inline assets/js/details.js]: Rename the variables in a camel case format while destructuring.

File: assets/js/main.js (hunk 11 above)
Comment (hedita) [reply assets/js/main.js]: Done.

File: assets/js/details.js (hunk 12 above)
Comment (hedita) [reply assets/js/details.js]: Done.

File: assets/js/details.js [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+   const {poster_path: posterPath, title, vote_average: voteAverage, overview} = await result.json();
+
+   document.body.style = ` background-image: linear-gradient(
+      var(--primary-gradient-color),
+      var(--secondery-gradient-color)),
+      url("https://image.tmdb.org/t/p/w342/${posterPath}");`
+
+  detailsContainer.innerHTML = `<img id="movie-img" class="img" src="https://image.tmdb.org/t/p/w342/${posterPath}" />
+   <figcaption class="details">
+     <h2 id="movie-name" class="movie-name"${title}></h2>
```
Comment (blacksrc) [inline assets/js/details.js]: fix the syntax here.

**Review by blacksrc — APPROVED** [review]
🚀 Well done!

**Pull request merged**